cucumber-expressions/java/src/main/java/io/cucumber/cucumberexpressions/TreeRegexp.java

```diff
@@ -24,13 +24,11 @@
 
     TreeRegexp(Pattern pattern) {
         this.pattern = pattern;
-        this.groupBuilder = createGroupBuilder(pattern);
+		this.groupBuilder = createGroupBuilder(pattern.pattern());
     }
 
-    private static GroupBuilder createGroupBuilder(Pattern pattern) {
-        String source = pattern.pattern();
-        Deque<GroupBuilder> stack = new ArrayDeque<>(singleton(new GroupBuilder()));
-        Deque<Integer> groupStartStack = new ArrayDeque<>();
+	public static GroupBuilder createGroupBuilder(String source) {
```

Comment:
We should consider moving this method to group so the api becomes `Group.from(Pattern)`.